lib/SimpleSAML/Utils/HTTP.php

```diff
@@ -327,8 +327,9 @@ public static function checkURLAllowed($url, array $trustedSites = null)
         }
         $url = self::normalizeURL($url);
 
-        if (filter_var($url, FILTER_VALIDATE_URL) === false) {
-            throw new Error\Exception('Invalid URL: '.$url);
+        // check the validity of the URL
+        if (parse_url($url,PHP_URL_HOST) === false) {
+            throw new \SimpleSAML_Error_Exception('Invalid URL: '.$url);
```

Comment:
Please note this is incorrect in `master`. The `SimpleSAML_Error_Exception` class has been migrated to namespaces, and therefore exceptions should be thrown as it is currently, by creating a new instance of `Error\Exception`.